collector/collector.go

```diff
@@ -0,0 +1,874 @@
+							Version:   PLUGIN_VERSION,
+						}
+						metrics = append(metrics, metric)
+					}
+				} else {
+					deviceID, err := strconv.Atoi(metricName[0])
+					if err != nil {
+						return nil, fmt.Errorf("In metric %s the given device id is invalid, err=%v", strings.Join(mt.Namespace.Strings(), "/"), err)
+					}
+					if deviceID > numOfDevices || deviceID < 0 {
+						return nil, fmt.Errorf("In metric %s the given device id is invalid, expected value in range 0-%d", strings.Join(mt.Namespace.Strings(), "/"), numOfDevices)
+					}
+
+					metric := plugin.Metric{
+						Timestamp: time.Now(),
+						Namespace: ns,
+						Data:      utils.GetValueByNamespace(c.containers[rid].Stats.Cgroups.BlkioStats.SectorsRecursive[deviceID], mt.Namespace.Strings()[len(ns)-1:]),
+						Tags:      mt.Tags,
+						Config:    mt.Config,
+						Version:   PLUGIN_VERSION,
+					}
+					metrics = append(metrics, metric)
+				}
+			} // the end of switch statsType
+		} // the end of range over ids
+	}
+
+	if len(metrics) == 0 {
+		return nil, fmt.Errorf("No metrics found")
+	}
+
+	return metrics, nil
+}
+
+// GetMetricTypes returns list of available metrics
+func (c *collector) GetMetricTypes(cfg plugin.Config) ([]plugin.Metric, error) {
+	var err error
+	var metricTypes []plugin.Metric
+
+	// initialize containerData struct
+	data := container.ContainerData{
+		Stats: container.NewStatistics(),
+	}
+
+	dockerMetrics := []string{}
+	utils.FromCompositeObject(data, "", &dockerMetrics)
+	nscreator := nsCreator{dynamicElements: definedDynamicElements}
+	for _, metricName := range dockerMetrics {
+		ns := plugin.NewNamespace(PLUGIN_VENDOR, PLUGIN_NAME).
+			AddDynamicElement("docker_id", "an id of docker container")
+
+		if ns, err = nscreator.createMetricNamespace(ns, metricName); err != nil {
+			// skip this metric name which is not supported
+			log.WithFields(log.Fields{
+				"block": "GetMetricTypes",
+			}).Warnf("Error in creating metric %s: err=%s", metricName, err)
+			continue
+		}
+		metricType := plugin.Metric{
+			Namespace: ns,
+			Version:   PLUGIN_VERSION,
+		}
+		metricTypes = append(metricTypes, metricType)
+	}
+
+	return metricTypes, nil
+}
+
+// GetConfigPolicy returns plugin config policy
+func (c *collector) GetConfigPolicy() (plugin.ConfigPolicy, error) {
+	policy := plugin.NewConfigPolicy()
+	configKey := []string{"intel", "docker"}
+
+	policy.AddNewStringRule(configKey,
+		"endpoint",
+		false,
+		plugin.SetDefaultString("unix:///var/run/docker.sock"))
+
+	policy.AddNewStringRule(configKey,
+		"procfs",
+		false,
+		plugin.SetDefaultString("/proc"))
+
+	return *policy, nil
+}
+
+type collector struct {
+	containers map[string]*container.ContainerData // holds data for a container under its short id
+	client     container.DockerClientInterface     // client for communication with docker (basic info, stats, mount points)
+	cgroupfs   string
+	driver     string
+	mounts     map[string]string
+	conf       map[string]string
```

Comment:
Coud you add comments to the rest of stucture's fields as it's done for "containers" and "client"?